consensus/hotstuff/timeout_collector.go

```diff
@@ -0,0 +1,50 @@
+package hotstuff
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// OnTCCreated is a callback which will be used by TimeoutCollector to submit a TC when it's able to create it.
+// This callback is called from processing logic.
+type OnTCCreated func(tc *flow.TimeoutCertificate)
+
+// OnPartialTCCreated is a callback which will be used by TimeoutCollector to notify about collecting f+1 timeouts for
+// some view.
+// This callback is called from processing logic.
+type OnPartialTCCreated func(view uint64)
+
+// OnNewQCDiscovered is a callback which will be called to notify about new validated QC.
+// This callback is called from processing logic.
+type OnNewQCDiscovered func(*flow.QuorumCertificate)
+
+// OnNewTCDiscovered is a callback which will be called to notify about new validated TC.
+// This callback is called from processing logic.
+type OnNewTCDiscovered func(*flow.TimeoutCertificate)
+
+// TimeoutCollector collects all timeout objects for a specified view. On the happy path, it
+// generates a TimeoutCertificate when enough timeouts have been collected.
+type TimeoutCollector interface {
```

Comment:
Would suggest to include a comment that implementations of `TimeoutCollector` are concurrency safe.